civo/provider.go

```diff
@@ -34,11 +37,11 @@ var (
 func Provider() *schema.Provider {
 	return &schema.Provider{
 		Schema: map[string]*schema.Schema{
-			"token": {
+			"credential_file": {
 				Type:        schema.TypeString,
 				Optional:    true,
-				DefaultFunc: schema.EnvDefaultFunc("CIVO_TOKEN", ""),
-				Description: "This is the Civo API token. Alternatively, this can also be specified using `CIVO_TOKEN` environment variable.",
```

Comment:
Instead of removing token, we can deprecate this field from the schema. This would be a better way.

```
			"token": {
				Type:        schema.TypeString,
				Optional:    true,
				DefaultFunc: schema.EnvDefaultFunc("CIVO_TOKEN", ""),
				Deprecated: "\"token\" attribute is deprecated. Moving forward, please use \"credential_file\" attribute.",
				Description: "This is the Civo API token. Alternatively, this can also be specified using `CIVO_TOKEN` environment variable.",
			},
```

Reply:
Hey @uzaxirr I zeroed in on the likely bottleneck,

The `token` variable in `provider.tf` in `build.sh` script was still there, I removed it.

Also updated the `provider.tf` in `examples` directory.

I also incorporated your suggestion of deprecating the token field. Thanks for the heads up.

Following are the screen grabs from local build:
<img width="718" alt="Screenshot 2024-07-09 172827" src="https://github.com/civo/terraform-provider-civo/assets/42026111/0ced17f1-49f8-48c5-9348-f2747f28b8b5">
<img width="833" alt="Screenshot 2024-07-09 172959" src="https://github.com/civo/terraform-provider-civo/assets/42026111/b851476f-acab-4066-851d-260a740f5d54">
<img width="811" alt="Screenshot 2024-07-09 173037" src="https://github.com/civo/terraform-provider-civo/assets/42026111/aed9d5b1-36fb-400f-aa87-454418a5a9d6">


The resource didn't get created, likely because I don't have an active Civo Account.
Can you now try again?

Regards.

Reply:
While you performed the above operation was there any `token` or credential_file attribute present in your provider.tf civo config block?

Reply:
@uzaxirr No, it was not in `provider.tf`. I will share you the terraform debug log which shows token being read from `~/.civo.json`. The code need some more changes and I am working on that.

Reply:
Sure, great work so far